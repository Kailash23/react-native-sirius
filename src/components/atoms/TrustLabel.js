import React from 'react';
import { View, Platform } from 'react-native';
import { Text, withTheme } from 'react-native-paper';
import { SIZES, getFontSize } from '@styles/sizes';
import { customFonts } from '@styles/customFonts';

function TrustLabel(props) {
  const { trustLabel } = props;
  const { colors } = props.theme;

  if (!trustLabel || !trustLabel.name) {
    return null;
  }
  return (
    <View
      style={{
        paddingBottom: 0,
        borderRadius: 2,
        justifyContent: 'center',
        alignItems: 'center',
        borderWidth: 0.7,
        borderColor: colors.onBackground
      }}>
      <Text
        style={{
          fontFamily:
            Platform.OS === 'ios'
              ? customFonts.merriweatherSansExtraBold
              : customFonts.merriweatherSansBold,
          fontWeight: Platform.OS === 'ios' ? 'bold' : '400',
          paddingHorizontal: 5,
          textTransform: 'uppercase',
          fontSize: getFontSize(SIZES.MINIFONT),
          justifyContent: 'center',
          textAlign: 'center',
          color: colors.secondaryText,
          paddingBottom: Platform.OS === 'android' ? 2 : 0
        }}>
        {trustLabel.name}
      </Text>
    </View>
  );
}

export default withTheme(TrustLabel);
