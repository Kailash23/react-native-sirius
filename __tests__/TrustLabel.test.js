import TrustLabel from '@atoms/TrustLabel';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

configure({ adapter: new Adapter() });

describe('Testing <TrustLabel/>', () => {
  test('TrustLabel should render properly', () => {
    const wrapper = shallow(
      <TrustLabel trustLabel={{ name: 'Test Trust Label' }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('TrustLabel should NOT return anything if the name is null ', () => {
    const wrapper = shallow(<TrustLabel trustLabel={{ name: null }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
