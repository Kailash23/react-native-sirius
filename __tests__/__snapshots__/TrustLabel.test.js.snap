// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TrustLabel/> TrustLabel should NOT return anything if the name is null  1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  trustLabel={
    Object {
      "name": null,
    }
  }
/>
`;

exports[`Testing <TrustLabel/> TrustLabel should render properly 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  trustLabel={
    Object {
      "name": "Test Trust Label",
    }
  }
/>
`;
